includes/payment-methods/class-upe-split-payment-gateway.php

```diff
@@ -548,6 +550,33 @@ public function get_stripe_id() {
 		return $this->stripe_id;
 	}
 
+	/**
+	 * Handles the shipping requirement for Afterpay payments.
+	 *
+	 * This method extracts the shipping and billing data from the order and sets the appropriate
+	 * shipping data for the Afterpay payment request. If neither shipping nor billing data is valid
+	 * for shipping, an exception is thrown.
+	 *
+	 * @param WC_Order                     $order    The order object containing shipping and billing information.
+	 * @param Create_And_Confirm_Intention $request The Afterpay payment request object to set shipping data on.
+	 *
+	 * @throws Invalid_Address_Exception If neither shipping nor billing address is valid for Afterpay payments.
+	 * @return void
+	 */
+	protected function handle_afterpay_shipping_requirement( WC_Order $order, Create_And_Confirm_Intention $request ): void {
```

Comment:
Good refactoring! Do we need to have it protected, or private is possible too?

Reply:
Modified to private in this commit https://github.com/Automattic/woocommerce-payments/pull/7541/commits/16c534233b8a75a7c727bd6ec870c87f91da4ea4 thanks!